bundles/org.openhab.binding.gardena/src/main/java/org/openhab/binding/gardena/internal/handler/GardenaAccountHandler.java

```diff
@@ -142,21 +147,23 @@ public void handleCommand(ChannelUID channelUID, Command command) {
     @Override
     public void onDeviceUpdated(Device device) {
         for (ThingUID thingUID : UidUtils.getThingUIDs(device, getThing())) {
-            Thing gardenaThing = getThing().getThing(thingUID);
-            try {
-                GardenaThingHandler gardenaThingHandler = (GardenaThingHandler) gardenaThing.getHandler();
-                gardenaThingHandler.updateProperties(device);
-                for (Channel channel : gardenaThing.getChannels()) {
-                    gardenaThingHandler.updateChannel(channel.getUID());
+            final Thing gardenaThing;
+            final GardenaThingHandler gardenaThingHandler;
+            if ((gardenaThing = getThing().getThing(thingUID)) != null
+                    && (gardenaThingHandler = (GardenaThingHandler) gardenaThing.getHandler()) != null) {
+                try {
+                    gardenaThingHandler.updateProperties(device);
+                    for (Channel channel : gardenaThing.getChannels()) {
+                        gardenaThingHandler.updateChannel(channel.getUID());
+                    }
+                    gardenaThingHandler.updateStatus(device);
+                } catch (GardenaException ex) {
+                    logger.error("There is something wrong with your thing '{}', please check or recreate it: {}",
+                            gardenaThing.getUID(), ex.getMessage());
+                    logger.debug("Gardena exception caught on device update.", ex);
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, ex.getMessage());
```

Comment:
Bindings should only log to error if something severe happened, like the detection of a bug in your code. This could be debug. See this link for a description of the log levels: https://www.openhab.org/docs/developer/guidelines.html#f-logging

The log messages could be removed, as the status update is already logged by the framework.

The state change originated by updateStatus() is logged to events.log. Including the status detail message.